src/VecSim/algorithms/hnsw/hnsw_batch_iterator.h

```diff
@@ -97,7 +99,7 @@ HNSW_BatchIterator<DataType, DistType>::scanGraph(candidatesMinHeap &candidates,
         this->candidates.empty()) {
```

Comment:
this one crashes when commenting  this->top_candidates_extras.empty(),(the batch iterator advanced test fails)
but works without this->candidates.empty()

Reply:
I think there was some corner case that required this check. I don't want to get into this part of the logic